pandas/io/excel/_base.py

```diff
@@ -977,9 +1007,9 @@ def save(self):
     def __init__(
         self,
         path: FilePath | WriteExcelBuffer | ExcelWriter,
-        engine=None,
-        date_format=None,
-        datetime_format=None,
+        engine: str | None = None,
```

Comment:
Could use literal + None

Reply:
Should probably add ExcelEngines in _typing.py

Reply:
I had a similar idea, but users can register custom writers like done in ``test_register_writer``, which would have different engine names. So I don't think we can restrict this here